frontend/src/views/admin/matching/import-export.tsx

```diff
@@ -0,0 +1,442 @@
+import { Modal, Button, Row, Form, Col } from "react-bootstrap";
+import { DataFormat } from "../../../libs/import-export";
+import { Match, AppointmentGuaranteeStatus } from "./types";
+import { upsertMatch, batchUpsertGuarantees, upsertNote } from "./actions";
+import { useThunkDispatch } from "../../../libs/thunk-dispatch";
+
+export function ExportMatchingDataButton() {
+    const matchingData = useSelector(matchingDataSelector);
+
+    function onClick() {
+        const blob = new Blob([JSON.stringify(matchingData)], {
+            type: "text/plain;charset=utf-8",
+        });
+        const date = new Date();
+        FileSaver.saveAs(
+            blob,
+            "matching_data-" +
+                date.getFullYear() +
+                "-" +
+                date.getMonth().toString().padStart(2, "0") +
+                "-" +
+                date.getDate().toString().padStart(2, "0") +
+                "-" +
+                date.getHours().toString().padStart(2, "0") +
+                "-" +
+                date.getMinutes().toString().padStart(2, "0") +
+                "-" +
+                date.getSeconds().toString().padStart(2, "0") +
+                ".json"
+        );
+    }
+
+    return (
+        <Button
+            variant="outline-primary"
+            size="sm"
+            className="footer-button"
+            onClick={onClick}
+        >
+            Export Data
+        </Button>
+    );
+}
+
+export function ImportMatchingDataButton({
+    markAsUpdated,
+}: {
+    markAsUpdated: Function;
+}) {
+    const [addDialogVisible, setAddDialogVisible] = React.useState(false);
+
+    function onClick() {
+        setAddDialogVisible(true);
+    }
+
+    const defaultLabel = "Select a JSON file.";
+
+    const [fileInputLabel, setFileInputLabel] = React.useState(defaultLabel);
+    const [fileArrayBuffer, setFileArrayBuffer] =
+        React.useState<ArrayBuffer | null>(null);
+    const [fileContent, setFileContent] = React.useState<DataFormat | null>(
+        null
+    );
+    const [diffedMatches, setDiffedMatches] = React.useState<Match[] | null>(
+        null
+    );
+    const [newGuarantees, setNewGuarantees] = React.useState<
+        AppointmentGuaranteeStatus[] | null
+    >(null);
+    const [newNotes, setNewNotes] = React.useState<
+        Record<string, string | null>
+    >({});
+
+    const matchingData = useSelector(matchingDataSelector);
+
+    function _onFileChange(event: React.ChangeEvent<HTMLInputElement>) {
+        if (!event.target || !event.target.files) {
+            return;
+        }
+        const file = event.target.files[0];
+        setFileInputLabel(file.name);
+
+        const reader = new FileReader();
+        reader.onload = (e) => {
+            if (!e.target || typeof e.target.result === "string") {
+                console.warn(
+                    "File of unexpected type",
+                    typeof e.target?.result
+                );
+                return;
```

Comment:
This error needs to be shown to the user. Either trigger an error popup or have error info trickle into the modal that this button is contained in.